File: js/shapes/square.js
const Shape = require("./shape");

class Square extends Shape {
  constructor(length = 10) {
    super("square");
    this.length = length;
  }

  calArea(borderThickness = 0) {
    return (
      (this.length + 2 * borderThickness) * (this.length + 2 * borderThickness)
    );
  }
}

module.exports = Square;

File: c#/shapes/Square.cs
namespace Canvas;

public class Square : Shape
{
    public int Length { get; set; }

    public Square(int length)
    {
        Type = "square";
        Length = length;
    }

    public override int CalArea(int borderThickness)
    {
        return (Length + 2 * borderThickness) * (Length + 2 * borderThickness);
    }

}
